src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -922,4 +923,58 @@ public String getMappedColumnName() {
         }
     }
 
+    /**
+     * This method is used to validate the dates of an exercise. A date is valid if there is no dueDateError or assessmentDueDateError
+     * @throws BadRequestException if the dates are not valid
+     */
+    public void validateDates() throws BadRequestException {
+        // All fields are optional, so there is no error if none of them is set
+        if (getReleaseDate() == null && getDueDate() == null && getAssessmentDueDate() == null) {
+            return;
+        }
+        // at least one is set, so we have to check the two possible errors
+        boolean validDates = validateDueDate(getReleaseDate(), getDueDate()) && validateAssessmentDueDate(getReleaseDate(), getDueDate(), getAssessmentDueDate());
+
+        if (!validDates) {
+            throw new BadRequestException();
+        }
+    }
+
+    /**
+     * This method is used to validate the dueDate of an exercise. A due date is valid, if it is after the releaseDate,
+     * @return true if there is no DueDateError
+     */
+    private boolean validateDueDate(ZonedDateTime releaseDate, ZonedDateTime dueDate) {
+        if (releaseDate == null) {
+            return true;
+        }
```

Comment:
I think this check is also performed as part of `isBeforeAndNotNull`? So this method can be removed altogether?